Sources/StreamChatUI/ChatMessageList/ChatThreadVC.swift

```diff
@@ -214,7 +214,13 @@ open class _ChatThreadVC<ExtraData: ExtraDataTypes>:
     }
     
     public var messages: [_ChatMessage<ExtraData>] {
-        messageController.replies + [messageController.message!]
+        let replies = Array(messageController.replies)
+        
+        if let threadRootMessage = messageController.message {
+            return replies + [threadRootMessage]
+        } else {
+            return replies
+        }
```

Comment:
Thread replies are evaluated from DTOs when converting `messageController.replies` to an array. The solution I was thinking of is appending thread root message to `replies: LazyCachedMapCollection` but seems it's not feasible 🤔 

To not break the indexes of lazy map we would have to modify the source collection of message DTOs and a cache. It's quite straightforward with cache but the information about source collection `Element` type is available only in `init` and doesn't get stored for later use.

Don't see how we can handle this nicely on UI side (on LLC side we could tweak a replies observer to also include a thread root though)